public/v1/daoyuan/js/couponEdit.js

```diff
@@ -54,18 +54,19 @@ $(document).ready(function(){
      rebateRule: rebateRule,
      valid: $('#update-valid').val()
     };
-    
     console.log(data);
     $.ajax({
       type: "PUT",
       url: putUrl,
-      data: data,
+      data: JSON.stringify(data),
+      contentType: "application/json",
       beforeSend: function(xhr)
       {
         xhr.setRequestHeader('x-access-token', getCookieByName('x-access-token'));
       },
       success: function(data, textStatus, xhr)
-      { 
+      {
+        debugger;
```

Comment:
Yes sure, just bear in mind, some of this potential bugs might be easier to handle while you are developing it. If you later come back to it, you might forget about it, since we are missing tests for front-end. But it's up to you. 

Also remember to remove some of the unnecessary debug from the server too.